tests/server/region_syncer/region_syncer_test.go

```diff
@@ -94,6 +94,26 @@ func (s *serverTestSuite) TestRegionSyncer(c *C) {
 		err = rc.HandleRegionHeartbeat(region)
 		c.Assert(err, IsNil)
 	}
+	// merge case
+	// region2 -> region1 -> region0
+	regions[0] = regions[0].Clone(core.WithEndKey(regions[2].GetEndKey()), core.WithIncVersion(), core.WithIncVersion())
+	err = rc.HandleRegionHeartbeat(regions[2])
+	c.Assert(err, IsNil)
+
+	// merge case
```

Comment:
do we need to add some comments about these cases?

Reply:
I have comment `region3 -> region4`, the direct is `->`. what your concerns?

Reply:
with `epoch`?

Reply:
PTAL